src/Solvers/Greedy.jl

```diff
@@ -78,7 +79,11 @@ function ssa_greedy_optimize(inputs, output, size, choose_fn=greedy_choose_simpl
     target_inds_histogram = histogram(Iterators.filter(∈(target_inds), Iterators.flatten(values(remaining))))
     high_ocurrent_inds = keys(filter(>(2) ∘ last, target_inds_histogram))
     # generate candidate pairwise contractions
-    queue = BinaryMinHeap{HeapNode{Int,NTuple{3,Set{Symbol}}}}()
+    if cost_fn == removedsize
+        queue = BinaryMinHeap{HeapNode{Int,NTuple{3,Set{Symbol}}}}()
+    else
+        queue = BinaryMinHeap{HeapNode{Float64,NTuple{3,Set{Symbol}}}}()
+    end
```

Comment:
This makes `queue` type unstable, which is bad for performance. Also, the type of `queue` being dependent on the `cost_fn` smells bad.

Reply:
This is only because the cost with the new  `removedsize_noise` function outputs a `Float`, instead of an `Int`. We could redefine `queue` to be more general and support both these possibilities.

Reply:
Let me review it further to decide this topic.

Reply:
Okay, replace it for `Float64`.
```suggestion
    queue = BinaryMinHeap{HeapNode{Float64,NTuple{3,Set{Symbol}}}}()
```